### DISCUSSION TITLE
Pagination

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#1384] [Feedback]
Use this place to discuss the [pagination component](https://ons-design-system.netlify.com/components/pagination) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
Some debugging variables have been left in the macro (https://github.com/ONSdigital/design-system/commit/b0fad430b71627150a96cb768780dbf3794c945f#diff-782048d2b2591ff1b22c6c6ebaf74303R17) and should be removed

**Comment by jrbarnes9** [comment]
Pager pagination would benefit from a bit more breathing space between itself and the content (IMHO)...

<img width="563" alt="Screenshot 2019-06-26 at 13 52 49" src="https://user-images.githubusercontent.com/43346934/60181375-f4ca1700-9819-11e9-955c-01cf51f53b5d.png">

**Comment by trevorsaint** [comment]
We need to remove the underline when the pager is in focus.
![image](https://user-images.githubusercontent.com/4989027/87150686-74eadf80-c2aa-11ea-8bc1-39306b16890f.png)

**Reply by jrbarnes9** [reply]
Resolved

**Comment by jrbarnes9** [comment]
Long pagination examples are broken

**Comment by trevorsaint** [comment]
We don’t need to specify 'false' on all items in the example docs.
Just add 'true' when required.

**Comment by jrbarnes9** [comment]
Typo in class name `ons-pageination__item--gap`

**Reply by jrbarnes9** [reply]
Fixed in #1740

**Comment by jrbarnes9** [comment]
- More space required between pagination position and items list. 
- Negative left margin on list needs to be removed when first page is current.
![Screenshot 2022-03-03 at 13 58 47](https://user-images.githubusercontent.com/43346934/156579578-81b49a5d-b005-4f3d-a00f-af11993a18be.png)

**Reply by jrbarnes9** [reply]
Raised #2024

**Comment by jrbarnes9** [comment]
I think the 'pager' should probably be documented separately unless we merge the macros.

**Reply by jrbarnes9** [reply]
Raised #2135

**Comment by jrbarnes9** [comment]
Raised #2178 regarding issue raised in https://github.com/ONSdigital/design-system/discussions/1385#discussioncomment-2552463

**Comment by jrbarnes9** [comment]
Pagination using the wrong colour variable for the active state.

Should be using `$color-text-link-active` (same as active section nav item) rather than `$color-text-link-hover`

**Reply by jrbarnes9** [reply]
#2430

**Comment by jrbarnes9** [comment]
Propose to align style of pagination and pager

From:
![Screenshot 2023-02-01 at 14 28 23](https://user-images.githubusercontent.com/43346934/216070408-7fa3c43c-168f-41d0-8ad6-a09533ce6d17.png)

To:
![Screenshot 2023-02-01 at 14 39 42](https://user-images.githubusercontent.com/43346934/216073691-08198d54-d1f2-4076-ada3-748cb40f83a6.png)

**Reply by jrbarnes9** [reply]
Raised #2598

**Comment by krisztin** [comment]
Hi 👋️, Accessibility/UX Dev here. Spotted a few issues with your pagination component:
- `aria-labels` are a bit too verbose. Putting "go to" into each one makes the screen reader experience unnecessarily protracted. People already expect a link to "go to" somewhere.
- On the current page link, the `aria-current` is set to `"true"` which is then announced as "link current" by screen readers. `"page"` would be a more appropriate value resulting in "page current".
- Missing `aria-hidden="true"` attribute from svgs on next and previous links to completely hide them from screen readers.
- Superfluous `role="img"` on the svgs. No need for accessible attributes on hidden elements.
- Superfluous `title` attributes on svgs. Titles in generally are not very helpful for accessibility (for instance, keyboard only users cannot access the values). The used value of `"ons-icon-arrow-next"` is especially unhelpful

**Reply by MagdalenaLarge** [reply]
Hello @krisztin ,
Thank you for getting in touch. We’re reviewing the issue and will follow up with you shortly.

**Reply by AlanHONS** [reply]
Confirm that related to current title= attributes on SVGs of Prev and Next arrow icons these are being flagged as a warning of failing WCAG2 Level A 4.1.2 by https://www.totalvalidator.com/ testing tool.

Here's screenshots from that tool against latest service manual example page https://service-manual.ons.gov.uk/design-system/components/pagination/example-pagination 
![image](https://github.com/user-attachments/assets/5a35209a-c1c5-4cb9-9796-f07f4d91fb82)

![image](https://github.com/user-attachments/assets/f67e5c72-1320-4135-af0e-2e0ed76c7f2c)

**Reply by MagdalenaLarge** [reply]
Thanks @AlanHONS ,
I will add your findings to our ticket.

**Reply by krisztin** [reply]
FYI, technically, in its current state the svgs fail [1.1.1 Non-text Content (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html) as they're non-text content that have an explicit role of `role="img"` which would require an accessible name. In reality, as this image does not add to the function of the _previous_ and _next_ buttons, as per 1.1.1 it should be implemented in a way that it is not accessible for assistive tech, hence the need to remove the `role="img"` and add `aria-hidden="true"`
